### ISSUE TITLE
Allow multiple backup locations per provider

### ISSUE THREAD

**Issue by ghost** [weaviate/weaviate#2977]
The current process for backup-s3 has an interface where a user passes in a UID to backup, then later on they can restore using the same UID. This however requires that the restore instance is running with the same BACKUP_S3_PATH otherwise the restore will return a 404 not found.

Users may want to isolate dev/stage/uat/prod environments with different BACKUP_S3_PATH, then after creating a backup in prod, it would be useful to restore dev/stage/uat with the same data for testing or dev work. Currently that does not appear to be possible unless you want to place all of your backups and restores for all environments in the same s3 path. You can modify your ids to encode the stage in them, but then all restores will appear in the same location and the files will have nothing indicating where the restore took place.

I think another solution would be a new interface which allows you to specify a complete path on s3 to the backup you want to use to restore, then it will use your BACKUP_S3_PATH when writing a backup_restore.json file and that json file can contain the key you restored. This will let a user quickly jump on to s3 and navigate to any stage and they can open any restore json file and see where the backup came from.

**Comment by WilliamWhispell** [comment]
I also tried manually copying from a prod path into a local path and on restore i saw:
{"action":"try_restore","backend":"s3","backup_id":"20230501_03_16_07_f6c7c","level":"error","msg":"node {\"node1\" \"172.18.0.4:7947\"}: cannot commit : corrupted backup file: invalid shard \"GoogleSearchResults\".\"dkfzCpvvvdAl\"","time":"2023-05-01T16:21:38Z","took":112181796}

So it looks like restores fail on different databases even if they have the same number of nodes, but a different shard name.

**Comment by trengrj** [comment]
I agree this feature would be useful. Renaming the issue to a more general "Allow multiple backup locations per provider".
